src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -288,5 +290,25 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 
             context.AddSource("GeneratedRouteBuilderExtensions.g.cs", code);
         });
+
+    }
+    private static void EmitHeaderValueParsing(CodeWriter codeWriter)
+    {
+        codeWriter.WriteLine("var headerParameters = handler.Method.GetParameters().Where(p => p.GetCustomAttribute<FromHeaderAttribute>() != null).ToList();");
```

Comment:
When emitting code at compile-time, we want to do the discovery of parameters like this via static analysis.

You'll observe that in other parts of the Request Delegate Generator (RDG) code, we generate an `EndpointParameter` object that we construct after statically analyzing the arguments then we use the information in that data model to emit the generated code. We don't need to re-compute things like the headerName dynamically like this.

You can find the pre-existing code where we emit the binding logic for header parameters [here](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs#L17).

Reply:
Thank you for the review, I will be working on this.